tools/shareable-examples/src/utils/get-example-module.ts

```diff
@@ -0,0 +1,163 @@
+  visitEachChild,
+  visitNode,
+  createIdentifier,
+  PropertyAssignment,
+  ImportDeclaration,
+  isSpreadElement,
+  createPropertyAssignment,
+} from 'typescript';
+import { ExampleAstFile } from './examples.interface';
+
+/**
+ * Creates a transformer to prepare the examples module, this includes the following:
+ * - remove all imports from the module that are relative, except the currently processed example
+ * - remove the DT_COMPONENT_EXAMPLES array declaration
+ * - remove the `entryComponents` from the module decorator
+ * - replace the `declarations` in the module decorator with the current example.
+ */
+export function createTransformer(
+  exampleRoot: string,
+  exampleClassName: string,
+): TransformerFactory<SourceFile> {
+  const transformer: TransformerFactory<SourceFile> = (
+    context: TransformationContext,
+  ) => {
+    function visitSpraedDeclarationAndReplaceIt(node: Node): Node | undefined {
+      if (isSpreadElement(node)) {
+        return createIdentifier(exampleClassName);
+      }
+      return visitEachChild(node, visitSpraedDeclarationAndReplaceIt, context);
+    }
+
+    /** Visit a propertyAssignment and determine which assignments need to be rewritten. */
+    function visitPropertyAssignment(
+      node: PropertyAssignment,
+    ): Node | undefined {
+      const assignmentName = node.name.getText();
+      // Replace the spread in the declarations with the one component
+      // we want to keep in there.
+      if (assignmentName === 'declarations') {
+        return visitEachChild(
+          node,
+          visitSpraedDeclarationAndReplaceIt,
+          context,
+        );
+      }
+      // Replace the entryComponents assignment with an exports,
+      // to be able to use the component outside of the declaring module.
+      if (assignmentName === 'entryComponents') {
+        const entryComponentsNode = visitEachChild(
+          node,
+          visitSpraedDeclarationAndReplaceIt,
+          context,
+        );
+        return createPropertyAssignment(
+          'exports',
+          entryComponentsNode.initializer,
+        );
+      }
+      return node;
+    }
+
+    /** Visit an importDeclaration and remove imports if necessary. */
+    function visitImportDeclaration(node: ImportDeclaration): Node | undefined {
+      const importFrom = node.moduleSpecifier.getText().replace(/'/g, '');
+      // By definition, our examples are nested each within its own folder
+      // we will need to keep the import for this one example, and all other
+      // imports, that are not nested in a folder here.
+      if (
+        // Check if it is a relative import
+        importFrom.startsWith('.') &&
+        // Check if it is not the one import we actually want to keep
+        !importFrom.includes(basename(exampleRoot)) &&
+        // Check if it is not any of the other relative imports we want to ditch.
+        importFrom.split('/').length > 2
+      ) {
+        return undefined;
```

Comment:
is the undefined necessary? would just a `return;` be the same?

Reply:
It would, I must have just blanked here because typescript complained about something. I have removed the undefined.